### ISSUE TITLE
Dropping also gives target element

### ISSUE THREAD

**Issue by hnry** [bevacqua/dragula#187]
Cool library, I would like to suggest a feature if I can.

Dropping an element onto a target container over a target container's element should report the target container's element too.

A scenario which this is useful is having the ability to stack elements (like cards). So a user drags to add / remove elements, but can also stack elements on top of each other. (The stacking effect not provided by dragula, but the additional API change is needed). Or getting a count, dragging element onto target element, check if elements are the same, then 'stack' (update count).

**Comment by neuropass** [comment]
+1

**Comment by beatfreaker** [comment]
Is passing `elementBehindCursor` (which will be target container or a div on which the dragged div is dropped) in this call :
```javascript
Before :: drake.emit('drop', item, target, _source);
After :: drake.emit('drop', item, target, _source, elementBehindCursor);
```
a good idea to achieve this?

@bevacqua what you think about this?

**Comment by bevacqua** [comment]
I think passing the `sibling` like we do on `accepts` is what they meant

**Comment by beatfreaker** [comment]
i guess @hnry is asking for feature to get target element on dropping which means on drop event right?

**Comment by hnry** [comment]
Either way sounds fine, either... directly getting the element that's being dropped on, or elements in between (because you can figure out which element got dropped on).


For instance imagine this example:
source -> `Shirt - Sweater - Pants`
target -> `Shirt - Sweater - Pants`

I drag from my source a Sweater on to a target Sweater (now I can write code to have 2 sweaters).
`Shirt - Sweater (2) - Pants`

So `.on('drop')` should provide a way for us to figure out where it was dropped within the target container itself. So either approach mentioned is fine as long as we can figure out where it was dropped.

I think `elementBehindCursor` is probably more accurate.

**Comment by beatfreaker** [comment]
@bevacqua what do you think?
if its fine i can make a PR for this, will pass `elementBehindCursor` in `.on('drop')` event.

**Comment by bevacqua** [comment]
I think we should make a PR where we add `sibling` to:

- `drop` event
- `moves` option

**Comment by beatfreaker** [comment]
@bevacqua you mean we could pass `_currentSibling` something like this :
`drake.emit('drop', item, target, _source,_currentSibling);`
right?

and what changes will be required for `moves` option?

**Comment by bevacqua** [comment]
On line 151 https://github.com/bevacqua/dragula/blob/master/dragula.js#L151 we should add nextEl(el) as the fourth parameter. We need documentation for these two changes as well.

**Comment by beatfreaker** [comment]
This doesn't work according to @hnry requirements see following demo :

![demo.gif](http://s15.postimg.org/yx2rc8giz/demo.gif)
see i am dropping the left hand side container's `div` on the first `div` of right hand side container so according to the requirement our function should return first `div` of right container as target element but as we are returning `sibling` it is returning the second `div` (here i am changing the `classname` of target div returned which is `sibling` ot `ex-moved` just to check what is getting returned)

@bevacqua what you think is it correct or not?

**Comment by bevacqua** [comment]
This is more consistent with what we're doing in `accepts`.

**Comment by beatfreaker** [comment]
Okay let me make a PR for this change then you check and let me know it is working fine or not.

**Comment by bevacqua** [comment]
Sounds good

**Comment by beatfreaker** [comment]
Check this [#204](https://github.com/bevacqua/dragula/pull/204) PR

**Comment by bevacqua** [comment]
Released as `3.3.0`
